### PR TITLE
Script file added to configure the test cases

### PR DESCRIPTION
**Pull request by 5380harshjain**
Here we are getting the database type and username from the pipeline as parameters,  and we are just validating the database type and running the script. Please review and let us know.

### CODE REVIEW

File: Scripts/Ticket_163723.ps1 [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿
+param (
+    [string] $dbType,
```
Comment (robe070) [inline Scripts/Ticket_163723.ps1]: This needs to be a list.

File: Scripts/Ticket_163723.ps1 [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    [string] $username
```
Comment (robe070) [inline Scripts/Ticket_163723.ps1]: Why do you need username?

File: Scripts/Ticket_163723.ps1 [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    Write-Output $line
+}
+#>
+
+
+$DbTestScript = "C:\Program Files (x86)\AZURESQL\LANSA\VersionControl\Scripts\db-test.ps1"
+
+
+if($file_data -eq $databaseType){
+    echo("Database Type is correct")
```
Comment (robe070) [inline Scripts/Ticket_163723.ps1]: Not echo. Use Write-Host. See post in Teams that I wrote on this. Very important!

File: Scripts/Ticket_163723.ps1 [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+#>
+
+$file_data = Get-Content "C:\Program Files (x86)\Lansa\Verifier_Connection.dat"
+
+echo($file_data)
+
+<#
+foreach($line in [System.IO.File]::ReadLines("C:\Program Files (x86)\Lansa\Verifier_Connection.dat"))
+{
+    Write-Output $line
```
Comment (robe070) [inline Scripts/Ticket_163723.ps1]: Write-Host! Otherwise you'll get messages coming out in the wrong order.

File: Scripts/Ticket_163723.ps1 [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+
+$DbTestScript = "C:\Program Files (x86)\AZURESQL\LANSA\VersionControl\Scripts\db-test.ps1"
+
+
+if($file_data -eq $databaseType){
+    echo("Database Type is correct")
+    &  "$DbTestScript" -compile $true -test $true
+}
+
```
Comment (robe070) [inline Scripts/Ticket_163723.ps1]: You are not going to find a match of database type with the lines in verifier_Connection.dat. There is a mapping. See the wiki or the Issue. Its somewhere.

**Review by robe070 — CHANGES_REQUESTED** [review]

File: Scripts/Ticket_163723.ps1 (hunk 5 above)
Comment (robe070) [reply Scripts/Ticket_163723.ps1]: The documentation on verifier_connection.dat is here:
https://github.com/lansa/db-regression/wiki#pipeline:~:text=The%20MSSQL%20configuration,execute%20the%20tests

File: Scripts/Ticket_163723.ps1 (hunk 2 above)
Comment (robe070) [reply Scripts/Ticket_163723.ps1]: @5380harshjain said:

> My understanding here is that we've to map the DB types with LU names in the script and then remove semicolons according to database type and lu name. So can we create key-value {dbType: LU_Name} pair in the script which will contain database type as key and lu_name as value and then iterate over all the database types which will be coming from pipeline and then check it with key-value pair and removing semicolons accordingly from the respective line. Is it the right approach? Please let me know.

Yes that sounds like the correct approach.

And I want to see Log Messages!

File: Scripts/Ticket_163723.ps1 (hunk 1 above)
Comment (5380harshjain) [reply Scripts/Ticket_163723.ps1]: Done

File: Scripts/Ticket_163723.ps1 (hunk 4 above)
Comment (5380harshjain) [reply Scripts/Ticket_163723.ps1]: Done

File: Scripts/Ticket_163723.ps1 (hunk 3 above)
Comment (5380harshjain) [reply Scripts/Ticket_163723.ps1]: Done.

File: Scripts/Ticket_163723.ps1 (hunk 2 above)
Comment (5380harshjain) [reply Scripts/Ticket_163723.ps1]: @robe070 I've made some changes, please review and let me know.

File: Scripts/Ticket_163723.ps1 (hunk 5 above)
Comment (5380harshjain) [reply Scripts/Ticket_163723.ps1]: Done.

File: Scripts/Ticket_163723.ps1 [hunk 6]
```diff
@@ -1,45 +1,52 @@
-﻿
+
+# dbTypes is containing database type names which will be coming from pipeline
 param (
-    [string] $dbType,
-    [string] $username 
+    [string[]] $dbTypes
```
Comment (robe070) [inline Scripts/Ticket_163723.ps1]: mandatory parameter

File: Scripts/Ticket_163723.ps1 [hunk 7]
```diff
@@ -1,45 +1,52 @@
...
+                $Content = $Content -replace $string,$string.Substring(1)
+            }
+        }
 
+        $Content | Set-Content -Path $Path
+    }
+}
 
+# DB test script 
+$DbTestScript = 'C:\Program Files (x86)\AZURESQL\LANSA\VersionControl\Scripts\db-test.ps1'
```
Comment (robe070) [inline Scripts/Ticket_163723.ps1]: This is only a true test if 'C:\Program Files (x86)\LANSA is the system being used. That's where Verifier_Connection.dat is.

And I presume you understand this is just a test. Running of the test should not happen here.

File: Scripts/Ticket_163723.ps1 [hunk 8]
```diff
@@ -1,45 +1,52 @@
...
-&  "$DbTestScript" -compile $true -test $true
-#>
+        $Path = 'C:\Program Files (x86)\Lansa\Verifier_Connection.dat'
+        $Content = [System.IO.File]::ReadAllLines($Path)
 
-#Start-Process powershell db-test.ps1 -compile $true -test $false
+        foreach ($string in (Get-Content 'C:\Program Files (x86)\Lansa\Verifier_Connection.dat'))
+        {
+            if($string.Contains($DB_LU_Map[$db])){
+                Write-Host("Changes made in $db")
```
Comment (robe070) [inline Scripts/Ticket_163723.ps1]: What changes? How does the reader of the log message understand what's been done?

You need a log message before here that says 'Configuring <full path to dat file>'

Then this message needs to be 'Enabling SuperServer test for $db'.

And you need to add a loop which first ensures that all connections HAVE a ';'.

**Review by robe070 — CHANGES_REQUESTED** [review]

File: Scripts/Ticket_163723.ps1 (hunk 6 above)
Comment (5380harshjain) [reply Scripts/Ticket_163723.ps1]: Done.

File: Scripts/Ticket_163723.ps1 (hunk 7 above)
Comment (5380harshjain) [reply Scripts/Ticket_163723.ps1]: Done.

File: Scripts/Ticket_163723.ps1 (hunk 8 above)
Comment (5380harshjain) [reply Scripts/Ticket_163723.ps1]: @robe070  I've made the mentioned changes, please review and let me know if you need anything else to be changed or anything I can improve in the script.

File: Scripts/configure-verifier-connection.ps1 [hunk 9]
```diff
@@ -0,0 +1,52 @@
+# dbTypes is containing database type names which will be coming from pipeline
+param (
+    [parameter(Mandatory=$true)]
+    [string[]] $dbTypes
```
Comment (robe070) [inline Scripts/configure-verifier-connection.ps1:3]: still not Mandatory

File: Scripts/configure-verifier-connection.ps1 [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+{
+    if($string.Substring(0, 1) -ne ";"){
+        $newString = ";" + $string
+        $Content = $Content -replace $string,$newString
+    }
+    $Content | Set-Content -Path $VerifierConnectionPath
+}
+
+
+Write-Host("Configuring... $VerifierConnectionPath")
```
Comment (robe070) [inline Scripts/configure-verifier-connection.ps1:32]: Just make the comment a log message. Replace all these lines:
```
Write-Host("Configuring... $VerifierConnectionPath")

#Iterarting over db types to remove the semicolon from particular line from verifier_connection.dat file
```
with
```
Write-Host("Iterating over db types to remove the semicolon from matching lines in $VerifierConnectionPath")
```

Thats what I said to do in the meeting.

File: Scripts/configure-verifier-connection.ps1 [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+#Iterarting over db types to remove the semicolon from particular line from verifier_connection.dat file
+foreach($db in $dbTypes){
+
+    if($DB_LU_Map.ContainsKey($db)){
+
+        foreach ($string in (Get-Content $VerifierConnectionPath))
+        {
+            if($string.Contains($DB_LU_Map[$db])){
+                Write-Host("Enabling SuperServer test for $db")
```
Comment (robe070) [inline Scripts/configure-verifier-connection.ps1]: Combine these. USE THE LOG MESSAGES AS COMMENTS! Thats what I said.
```
                Write-Host("Enabling SuperServer test for $db by removing the semicolon")
```

**Review by robe070 — CHANGES_REQUESTED** [review]

File: Scripts/configure-verifier-connection.ps1 (hunk 9 above)
Comment (robe070) [reply Scripts/configure-verifier-connection.ps1:3]: I'm blind! I'm used to it on the same line. Its fine as it is.

Comment (5380harshjain) [reply Scripts/configure-verifier-connection.ps1:3]: oh, no issues. Thanks @robe070

**Comment by robe070** [comment]
>  I've made the mentioned comments as log messages, but do we have to write any comments? I was writing it just for reference, currently I've removed all the extra comments.

Almost anything that might be a comment should be a Write-Host.
That is write log messages instead of comments.
Why double up?
Its seems clear to me. I don't understand your confusion.

**Comment by 5380harshjain** [comment]
> > I've made the mentioned comments as log messages, but do we have to write any comments? I was writing it just for reference, currently I've removed all the extra comments.
> 
> Almost anything that might be a comment should be a Write-Host. That is write log messages instead of comments. Why double up? Its seems clear to me. I don't understand your confusion.

I was writing the comments for developer reference for the future, but I've changed the mentioned comments into logs and removed the others not required.

File: Scripts/configure-verifier-connection.ps1 (hunk 10 above)
Comment (5380harshjain) [reply Scripts/configure-verifier-connection.ps1:32]: Done.

File: Scripts/configure-verifier-connection.ps1 (hunk 11 above)
Comment (5380harshjain) [reply Scripts/configure-verifier-connection.ps1]: Done.

**Pull request merged**